### ISSUE TITLE
Logout > Login sometimes restores previous credentials

### ISSUE THREAD

**Issue by tristan-warner-smith** [magiclabs/magic-ios#34]
**Describe the bug**
After calling the log out API and subsequently attempting to log in again with a different email address, we're being logged in with the previous credentials not the ones entered.

**To Reproduce**
In code terms we're calling:
```
try await magic.user.getIdToken().async() // which throws
try await magic.auth.loginWithEmailOTP(.init(email: "firstEmail@test.com")).async()
// Then we log out
try await magic.user.logout().async()
// Then we go to log in with a different email address but
try await magic.user.getIdToken().async() // returns the previous token
// Which means they're logged in but with the previous email
```
* As above

**Expected behavior**
Once you've called the `logOut()` method, subsequent calls to `getIdToken` will throw.

**Environment**
 - Package Version: 10.0.0
 - Xcode Version: 15.2, 15.3 beta
 - Swift Version: 5.9, 5.10
 - Simulator / Device OS: * Simulator + Device (iOS 17.2.1, iOS 17.4)

**Additional context**
Log out is an unusual operation in our app but this issue is subjectively reproducible 15-25% of the time in our testing.

**Comment by Ariflo** [comment]
Hi Tristan, Thank you for bringing this to our attention. We are in the middle of triaging it and will address the issue shortly!

**Comment by Ariflo** [comment]
@tristan-warner-smith quick update, both issues you've brought up will be addressed before the end of  next week. Thank you for your patience.

**Comment by tristan-warner-smith** [comment]
> @tristan-warner-smith quick update, both issues you've brought up will be addressed before the end of next week. Thank you for your patience.

Hey @Ariflo, I'm guessing the credential caching-issue turned out to be trickier than expected. Are they still on track for a release this week?

**Comment by Ariflo** [comment]
@tristan-warner-smith yes, apologies for the delay. We're working on getting this fix up and out asap!

**Comment by Ariflo** [comment]
@tristan-warner-smith we're glad to hear you were able to resolve your caching issues. We're working on getting our iOS SDKs and demo apps up-to-date with the latest versions of Swift. We'll make sure to note this down on the README to avoid further confusion. Thanks!

**Comment by tristan-warner-smith** [comment]
@Ariflo unfortunately I spoke too soon, can you reopen this issue?
The behaviour we're seeing is that the logout call returns success, we see it in the response returned but the subsequent call to `user.getIdToken` returns a token, not `nil` like it should.

**Comment by tristan-warner-smith** [comment]
@Ariflo I spent some time looking at the iOS code that's visible and, although I can't see the full javascript or the loaded webpage and can only guess what is causing this caching behaviour. I wonder if you've explored clearing  `WKWebsiteDataStore` or `HTTPCookieStorage.shared`.

As I mentioned this is significantly impacting our development and sign-off processes so anything we can do to help / speed the resolution up, let me know.

**Comment by tristan-warner-smith** [comment]
@Ariflo calling this to clear the cache after logout seems to have done the trick, so you might want to do something similar.

```swift
@MainActor
func clearMagicWebKitDataStore() async {
    Logger.auth.info("Checking for Magic link web cache")

    await withCheckedContinuation { continuation in
        let store = WKWebsiteDataStore.default()
        store.fetchDataRecords(ofTypes: WKWebsiteDataStore.allWebsiteDataTypes()) { records in
            guard let magicRecord = records.first(where: { $0.displayName == "magic.link"}) else {
                debugPrint("Magic link had nothing cached")
                return continuation.resume()
            }
            debugPrint("Clearing magic link cache")
            store.removeData(
                ofTypes: WKWebsiteDataStore.allWebsiteDataTypes(),
                for: [magicRecord],
                completionHandler: {
                    debugPrint("Magic link cache cleared")
                    continuation.resume()
                }
            )
        }
    }
}
```

**Comment by tristan-warner-smith** [comment]
I can confirm @Ariflo that this fixes the auth caching issue we mentioned. Though I imagine you'll want to tackle the core issue of whatever is being cached incorrectly with less of a sledgehammer approach.
